### PR TITLE
*: support unix domain socket

### LINKED ISSUE #296: add UNIX domain socket support
Support for using unix domain socket would be nice.

### PR DESCRIPTION
**Pull request by BusyJay**
It's supported by c core, we just need to allow it as parameter.

Close #296.

### CODE REVIEW

File: tests-and-examples/tests/cases/misc.rs [hunk 1]
```diff
@@ -129,3 +129,38 @@ fn test_soundness() {
     }
     assert_eq!(counter.load(Ordering::SeqCst), 9000);
 }
+
+#[test]
+fn test_uds() {
```
Comment (nrc) [inline tests-and-examples/tests/cases/misc.rs]:
```suggestion
fn test_unix_domain_service() {
```

**Comment by BusyJay** [comment]
There should be bugs with rustc nightly. I can compile the PR with last nightly version.

**Comment by BusyJay** [comment]
@nrc PTAL

File: src/server.rs [hunk 2]
```diff
@@ -64,7 +64,9 @@ where
 /// Given a host and port, creates a string of the form "host:port" or
 /// "[host]:port", depending on whether the host is an IPv6 literal.
 fn join_host_port(host: &str, port: u16) -> String {
-    if let Ok(ip) = host.parse::<IpAddr>() {
+    if host.starts_with("unix:") {
```
Comment (hicqu) [inline src/server.rs]: The return value must be nul-terminated, so how about use `CString` instead?

Comment (BusyJay) [reply src/server.rs]: There may be two allocations, one is for format, second is for cstring.

**Pull request merged**